prog/base.rb

```diff
@@ -9,6 +9,13 @@ def initialize(strand, snap = nil)
     @subject_id = frame.dig("subject_id") || @strand.id
   end
 
+  # Searches the stack for the Prog that caused execution of the code,
+  # which can be useful in logging from nested method calls.
+  def self.current_prog
+    $~ = nil
+    caller.rindex { it[/Prog::(\w+)#(\w+)/] } && $&
```

Comment:
Giving this some more thought, I think this would be less objectionable:


```suggestion
    caller_locations.reverse_each { return it.label if it.label.start_with?("Prog::") }
    nil
```

Parsing entire backtrace lines is brittle and less efficient.  Additionally, the suggested works for progs that have `::` in them (`\w` doesn't match `:`).